### PR TITLE
Android: Update APSAnalytics

### PR DESCRIPTION
**Pull request by garymathews**
- Include @whitfin's Titanium SDK changes after refactoring APSAnalytics
- Fix `Ti.Platform.architecture` and `Ti.Platform.ostype`
- Update `aps-analytics.jar`

**Test code sample:**
_app.js_
```
var win = Ti.UI.createWindow({layout: 'vertical'}),
	button1 = Ti.UI.createButton({title: 'nav - no'}),
	button2 = Ti.UI.createButton({title: 'nav - yes'}),
	button3 = Ti.UI.createButton({title: 'nav - offline'}),
	button4 = Ti.UI.createButton({title: 'feat - no'}),
	button5 = Ti.UI.createButton({title: 'feat - yes'}),
	button6 = Ti.UI.createButton({title: 'feat - offline'}),
	button7 = Ti.UI.createButton({title: 'opt out/in'});

button1.addEventListener('click', function () {
	Ti.Analytics.navEvent('from', 'to', 'nav event opted out');
});
button2.addEventListener('click', function () {
	Ti.Analytics.navEvent('from', 'to', 'nav event not opted out');
});
button3.addEventListener('click', function () {
	Ti.Analytics.navEvent('from', 'to', 'nav event offline');
});
button4.addEventListener('click', function () {
	Ti.Analytics.featureEvent('feat no');
});
button5.addEventListener('click', function () {
	Ti.Analytics.featureEvent('feat yes');
});
button6.addEventListener('click', function () {
	Ti.Analytics.featureEvent('feat offline');
});
button7.addEventListener('click', function() {
	if (Ti.Analytics.getOptedOut()) {
		Ti.Analytics.optedOut = false;
		alert('Opted out set to FALSE');
	} else {
		Ti.Analytics.setOptedOut(true);
		alert('Opted out set to TRUE');
	}
});
win.add([button1, button2, button3, button4, button5, button6, button7])
win.open();
```

### CODE REVIEW

File: android/modules/platform/src/java/ti/modules/titanium/platform/PlatformModule.java [hunk 1]
```diff
@@ -228,7 +237,44 @@ public boolean openURL(String url)
 	public String getMacaddress()
 	// clang-format on
 	{
-		return TiPlatformHelper.getInstance().getMacaddress();
+		//		String macaddr = null;
```
Comment (garymathews) [inline android/modules/platform/src/java/ti/modules/titanium/platform/PlatformModule.java]: We should deprecate this, on Android 6.0+ the MAC address will always return `02:00:00:00:00:00`

**Comment by sgtcoolguy** [comment]
Closing as per @garymathews

**Comment by ypbnv** [comment]
Reopened for combining with the refactored aps_analytics library.

**Comment by ypbnv** [comment]
Pushed the aps-analytics.jar version containing optOut option and refactoring.

**Comment by garymathews** [comment]
Up-to date with aps_sdk, merging.

**Pull request merged**